hadoop-connector-examples/src/main/java/io/pravega/example/hadoop/PravegaFixedSegmentsOutputFormat.java

```diff
@@ -55,51 +55,53 @@
  */
-public class PravegaOutputFormat<V> extends OutputFormat<String, V> {
+public class PravegaFixedSegmentsOutputFormat<V> extends OutputFormat<String, V> {
 
-    private static final Logger log = LoggerFactory.getLogger(PravegaOutputFormat.class);
+    private static final Logger log = LoggerFactory.getLogger(PravegaFixedSegmentsOutputFormat.class);
 
     // Pravega scope name
     public static final String SCOPE_NAME = "pravega.scope";
     // Pravega stream name
-    public static final String OUT_STREAM_NAME = "pravega.out.stream";
+    public static final String STREAM_NAME = "pravega.stream";
+    // Pravega stream segments
+    public static final String STREAM_SEGMENTS = "pravega.stream.segments";
     // Pravega uri string
     public static final String URI_STRING = "pravega.uri";
     // Pravega deserializer class name
     public static final String DESERIALIZER = "pravega.deserializer";
 
-    private static final long DEFAULT_TXN_TIMEOUT_MS = 30000L;
-    private static final long DEFAULT_TXN_MAX_EXECUTION_TIME_MS = 30000L;
-    private static final long DEFAULT_PING_LEASE_MS = 30000L;
+    static final long DEFAULT_TXN_TIMEOUT_MS = 30000L;
 
     // client factory
     private ClientFactory externalClientFactory;
 
-    public PravegaOutputFormat() {
+    public PravegaFixedSegmentsOutputFormat() {
     }
 
     @VisibleForTesting
-    protected PravegaOutputFormat(ClientFactory externalClientFactory) {
+    protected PravegaFixedSegmentsOutputFormat(ClientFactory externalClientFactory) {
         this.externalClientFactory = externalClientFactory;
     }
 
     @Override
     public RecordWriter<String, V> getRecordWriter(TaskAttemptContext context) throws IOException, InterruptedException {
+
         Configuration conf = context.getConfiguration();
-        final String scopeName = Optional.ofNullable(conf.get(PravegaOutputFormat.SCOPE_NAME)).orElseThrow(() ->
-                new IOException("The input scope name must be configured (" + PravegaOutputFormat.SCOPE_NAME + ")"));
-        final String streamName = Optional.ofNullable(conf.get(PravegaOutputFormat.OUT_STREAM_NAME)).orElseThrow(() ->
-                new IOException("The input stream name must be configured (" + PravegaOutputFormat.OUT_STREAM_NAME + ")"));
-        final URI controllerURI = Optional.ofNullable(conf.get(PravegaOutputFormat.URI_STRING)).map(URI::create).orElseThrow(() ->
-                new IOException("The Pravega controller URI must be configured (" + PravegaOutputFormat.URI_STRING + ")"));
-        final String deserializerClassName = Optional.ofNullable(conf.get(PravegaOutputFormat.DESERIALIZER)).orElseThrow(() ->
-                new IOException("The event deserializer must be configured (" + PravegaOutputFormat.DESERIALIZER + ")"));
+        final String scopeName = Optional.ofNullable(conf.get(SCOPE_NAME)).orElseThrow(() ->
+                new IOException("The input scope name must be configured (" + SCOPE_NAME + ")"));
```

Comment:
I know that this is not something that this PR is directly changing, but incorrect configuration should be a runtime exception and not an IOException. I'm not sure that would violate the way Hadoop handles exceptions and cause problems to applications.

In any case, I'm raising it not because I'm requesting that we change it in this PR, but only to call you attention to it. If you agree, the best course of action would be to create a separate issue.

Reply:
Yes, https://github.com/pravega/connectors-loadtest/issues/3 created to address the issues

Reply:
Yes it is indeed due to the API defined by MapReduce: 
public abstract RecordWriter<K, V> getRecordWriter(TaskAttemptContext var1) throws IOException, InterruptedException;